### PR TITLE
Remove duplicated ruleLabels in DerivedFromRules

### PR DESCRIPTION
**Pull request by aanm**
Read per commit

### CODE REVIEW

**Comment by aanm** [comment]
test-me-please

File: pkg/policy/l4.go [hunk 1]
```diff
@@ -786,7 +786,16 @@ func addL4Filter(policyCtx PolicyContext,
 		filterToMerge.detach(selectorCache)
 		return err
 	}
-	existingFilter.DerivedFromRules = append(existingFilter.DerivedFromRules, ruleLabels)
+	var exists bool
+	for _, existingRules := range existingFilter.DerivedFromRules {
```
Comment (jrajahalme) [inline pkg/policy/l4.go]: I'd have "labels" somewhere in the variable name, like `existingRuleLabels` to make it clearer we not somehow comparing a "rule" to "labels" below.

**Review by jrajahalme — APPROVED** [review]
Small nit only.

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
@jrajahalme I had to fix a couple of unit tests, might be worth taking a 2nd look just to make sure I didn't miss anything.

**Pull request merged**